milestones/api.py

```diff
@@ -39,6 +39,16 @@ def _validate_content_key(content_key):
         )
 
 
+def _validate_requirements(requirements):
```

Comment:
It's not clear from the name of this method if it is specific to CourseContentMilestone validation -- we should probably rename it, or make the method more general/flexible by passing in the type being validated so 'validators' can react accordingly.